main.go

```diff
@@ -178,6 +185,19 @@ func CreateExecutionPlan(cfg *Config) ([]string, error) {
 			args = append(args, "--set-env-vars", envStr)
 		}
 
+		if len(cfg.Secrets) > 0 {
+			e := make([]string, 0)
+			for k, v := range cfg.Secrets {
+				e = append(e, fmt.Sprintf(`%s=%s`, k, v))
+			}
+
+			// we're using ":||:" as the separator for the args, let's hope no one puts that in an env variable value
+			sep := ":||:"
```

Comment:
nit: you could move this separator out in a const to the top of the file and use it in both places.